src/mindlessgen/prog/config.py

```diff
@@ -735,6 +737,61 @@ def __init__(self, config_file: str | Path | None = None):
         if config_file:
             self.load_from_toml(config_file)
 
+    def check_config(self):
+        """
+        Checks ConfigClass for any incompatibilities that are imaginable
+        """
+        # lower number of the available cores and the configured parallelism
+        num_cores = min(mp.cpu_count(), self.general.parallel)
+        if self.general.parallel > mp.cpu_count():
+            warnings.warn(
+                f"Number of cores requested ({self.general.parallel}) is greater "
+                + f"than the number of available cores ({mp.cpu_count()})."
+                + f"Using {num_cores} cores instead."
+            )
+        if self.general.verbosity > 0:
+            print(f"Running with {num_cores} cores.")
+
+        if num_cores > 1 and self.general.verbosity > 0:
+            # raise warning that parallelization will disable verbosity
+            warnings.warn(
+                "Parallelization will disable verbosity during iterative search. "
+                + "Set '--verbosity 0' or '-P 1' to avoid this warning, or simply ignore it."
+            )
+        if num_cores > 1 and self.postprocess.debug:
+            # raise warning that debugging of postprocessing will disable parallelization
+            warnings.warn(
+                "Debug output might seem to be redundant due to the parallel processes "
+                + "with possibly similar errors in parallel mode. "
+                + "Don't be confused!"
+            )
+
+        # Check for f-block elements in forbidden elements
+        if self.generate.forbidden_elements:
+            f_block_elements = set(range(56, 70)) | set(range(88, 102))
+            if any(
+                elem not in f_block_elements
+                for elem in self.generate.forbidden_elements
+            ):
+                warnings.warn(
+                    "f-block elements could be within the molecule. xTB does not treat f electrons explicitly. In this case UHF is set to 0."
+                )
+
+        # Check for super heavy elements in forbidden elements
+        super_heavy_elements = set(range(86, 102))
+        if self.generate.element_composition and any(
+            elem in super_heavy_elements for elem in self.generate.element_composition
+        ):
+            warnings.warn(
+                "Super heavy elements are within the molecule. xTB does not treat super heavy elements. Atomic numbers are reduced by 32."
```

Comment:
Rewrite in such a way that it's clear that the `MindlessGen` generation DOES still work but in a kinda approximated fashion.